java/src/main/native/src/row_conversion.cu

```diff
@@ -835,6 +845,62 @@ copy_validity_to_rows(const size_type num_rows, const size_type num_columns,
   }
 }
 
+/**
+ * @brief kernel to copy string data to JCUDF row format
+ *
+ * @tparam RowOffsetIter iterator for row offsets into the destination data
+ * @param num_rows number of rows in this portion of the table
+ * @param num_variable_columns number of columns of variable-width data
+ * @param variable_input_data variable width data column pointers
+ * @param variable_col_output_offsets output offset information for variable-width columns
+ * @param variable_col_offsets input offset information for variable-width columns
+ * @param fixed_width_row_size offset to variable-width data in a row
+ * @param row_offsets offsets for each row in output data
+ * @param batch_row_offset row start for this batch
+ * @param output_data pointer to output data for this batch
+ *
+ */
+template <typename RowOffsetIter>
+__global__ void copy_strings_to_rows(size_type const num_rows, size_type const num_variable_columns,
+                                     int8_t const **variable_input_data,
+                                     size_type const *variable_col_output_offsets,
+                                     size_type const **variable_col_offsets,
+                                     size_type fixed_width_row_size, RowOffsetIter row_offsets,
+                                     size_type const batch_row_offset, int8_t *output_data) {
+  // Each block will take a group of rows controlled by NUM_STRING_ROWS_PER_BLOCK.
+  // Each warp will copy a row at a time. The base thread will first go through column data and
```

Comment:
> // Each warp will copy a row at a time.

Forgive me, but each block or each thread?

Reply:
This is per block. All threads in a warp act on a single row to copy all the string data.

Reply:
Sorry, I meant to resolve this, after I had read the rest of the code.